File: DrawPatterns/Patterns.py
print("Patterns")
print()
print("-#-" * 20)
print()

# Triangle 1
def triangle1(lines, columns):
    print(" -- Triangle 1 -- ")
    for i in range(lines):
        for j in range(columns):
            if j < i:
                print(" ", end="")
            else:
                print("X ", end="")
        print("")
    print()

# Triangle 2
def triangle2(lines, columns):
    print(" -- Triangle 2 -- ")
    for i in range(lines):
        for j in range(columns, -1, -1):
            if j > i:
                print(" ", end="")
            else:
                print("X ", end="")
        print("")
    print()

# Triangle 3
def triangle3(lines, columns):
    print(" -- Triangle 3 -- ")
    for i in range(lines):
        for j in range(columns):
            if j > i:
                print(" ", end="")
            else:
                print("X ", end="")
        print("")
    print()

# Triangle 4
def triangle4(lines, columns):
    print(" -- Triangle 4 -- ")
    for i in range(lines, -1, -1):
        for j in range(columns):
            if j >= i:
                print(" ", end="")
            else:
                print("X ", end="")
        print("")
    print()


# Box
def box(lines,columns):
    print(" -- Box -- ")
    for i in range(0, lines + 1):
        for j in range(0, columns + 1):
            if i == 0 or i == lines or j == 0 or j == lines:
                print("X ", end="")
            else:
                print("  ", end="")
        print()
    print()

# Box X
def boxX(lines, columns):
    print(" -- Box X -- ")
    for i in range(0, lines + 1):
        for j in range(0, columns + 1):
            if i == 0 or i == lines or j == 0 or j == lines or i == j or i + j == columns:
                print("X ", end="")
            else:
                print("  ", end="")
        print()
    print()

triangle1(10, 10)
triangle2(10, 10)
triangle3(10, 100)
triangle4(10, 10)

box(4, 4)
boxX(10, 10)

File: DrawPatterns/Patterns.js
function box(lines = 10, columns = 10) {
	let result = "";
	for (let i = 0; i < lines; i++) {
		for (let j = 0; j < columns; j++) {
			if (i === 0 || i === lines - 1 || j === 0 || j === columns - 1) result += "X"
			else result += " "
		}
		result += "\n";
	}
	return result;
}

function boxX(lines = 10, columns = 10) {
	let result = "";
	for (let i = 0; i < lines; i++) {
		for (let j = 0; j < columns; j++) {
			if (i === 0 || i === lines - 1 ||
				j === 0 || j === columns - 1 ||
				i === j || i + j === lines) result += "X"
			else result += " "
		}
		result += "\n";
	}
	return result;
}

function xLines(lines = 10, columns = 10) {
	let result = "";
	for (let i = 0; i < lines; i++) {
		for (let j = 0; j < columns; j++) {
			if (i === j || i + j === lines) result += "X"
			else result += " "
		}
		result += "\n";
	}
	return result;
}

function triangle1(lines = 10) {
	let result = "";
	for (let i = 0; i < lines; i++) {
		for (let j = 0; j < i; j++) {
			result += "X";
			if (j < i - 1) result += " ";
		}
		result += "\n";
	}

	return result;
}

function triangle2(lines = 10) {
	let result = "";
	for (let i = lines; i > 0; i--) {
		for (let j = i; j > 0; j--) {
			result += "X";
			if (j > 1) result += " ";
		}
		result += "\n";
	}
	return result;
}

function triangle3(lines = 10) {
	let result = "";
	for (let i = lines; i > 0; i--) {
		for (let j = 0; j < lines; j++) {
			if (j > i) result += "X";
			result += " ";
		}
		result += "\n";
	}
	return result;
}

function triangle4(lines = 10) {
	let result = "";
	for (let i = 0; i < lines; i++) {
		for (let j = 0; j < lines; j++) {
			if (j > i) result += "X";
			result += " ";
		}
		result += "\n";
	}
	return result;
}

console.log(box(10, 10));
console.log(boxX(15, 15));
console.log(xLines(15, 20));

console.log(triangle1(20));
console.log(triangle2(20));
console.log(triangle3(20));
console.log(triangle4(20));
